tests/controller/controller_test.go

```diff
@@ -510,12 +512,23 @@ func TestBackendSecurityPolicyController(t *testing.T) {
 		require.NoError(t, err)
 
 		require.Eventually(t, func() bool {
+			secretName := rotators.GetBSPSecretName(backendSecurityPolicyName)
 			var bsp aigv1a1.BackendSecurityPolicy
 			err = c.Get(t.Context(), client.ObjectKey{Name: backendSecurityPolicyName, Namespace: backendSecurityPolicyNamespace}, &bsp)
 			if err == nil || client.IgnoreNotFound(err) != nil {
 				t.Logf("expected not found error, got: %v", err)
 				return false
 			}
+			var secret corev1.Secret
+			errSecret := c.Get(t.Context(), client.ObjectKey{
+				Name:      secretName,
+				Namespace: backendSecurityPolicyNamespace,
+			}, &secret)
+
+			if !apierrors.IsNotFound(errSecret) {
+				t.Logf("expected secret %s/%s to be deleted, but still exists", backendSecurityPolicyNamespace, secretName)
+				return false
+			}
```

Comment:
can you check the secret is also correctly created inside `t.Run("update security policy", func(t *testing.T) {` above? otherwise this assertion becomes useless

Reply:
it might be hard to test this as this may require a real OIDC server working. You can delete the change in this file altogether if you like